code/ship/ship.cpp

```diff
@@ -10360,11 +10360,10 @@ void update_firing_sounds(object* objp, ship* shipp)
 				end_snd_played = wip->end_firing_snd;
 			}
 
-			if (wip->loop_firing_snd.isValid()) {
+			if (swp->firing_loop_sounds[i] != -2) {
```

Comment:
This should probably be `> -1`. That's already been checked for, but it makes the logic here in a vacuum a bit clearer: only call `obj_snd_delete` when there's an actual valid index in there to stop.